apps/full-site-editing/full-site-editing-plugin/block-helpers.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * External dependencies
+ */
+import { getCategories } from '@wordpress/blocks';
+
+/**
+ * Indicates whether or not we're in an environment that is still
+ * using old block [category names]{@link https://github.com/WordPress/gutenberg/pull/19279}.
+ * This is not perfect, but works well for checking if we need to register
+ * a block with an old category name or the new one, in order to provide
+ * backwards compatibility with older Gutenberg versions.
+ *
+ * This is the same strategy used by [CoBlocks]{@link https://github.com/godaddy-wordpress/coblocks/pull/1535}.
+ *
+ * @returns {boolean} value indicating if we're still using old category names.
+ */
+export const hasLegacyCategory = getCategories().some( function ( category ) {
+	return category.slug === 'formatting';
+} );
```

Comment:
Instead of a capability check you use to build conditionals, we wrap the conditional logic and accept a list of candidate categories?

```ts
function getCategoryWithFallbacks( ...requestedCategories: string[]): string {
	const knownCategories: Array< { slug: string } > = getCategories();
	for ( const requestedCategory of requestedCategories) {
		if ( knownCategories.some( ( { slug } ) => slug === requestedCategory ) ) {
			return requestedCategory;
		}
	}
	// Should we throw an error here? 🤔
	throw new Error('Terrible choice of category 🤦‍♀️');
}

// Usage:
registerBlockType( 'my-block/name', { category: getCategoryWithFallbacks( 'layout', 'widgets' ) } );
```

Reply:
That's a nice idea! Avoiding having to write a ternary expression would be nice, indeed. Just wondering how we could make this function re-usable across all packages that register blocks. We'll probably need to duplicate it for now?

Reply:
Yep, we'll need to write it once for each package (FSE, Jetpack…).

Reply:
Btw, I really like this approach as it's basically future proof. If we change categories again, then it's just a matter of passing it as the last one in the argument.

Reply:
I'm wondering if it makes sense to move this function over to `@wordpress/blocks`... :thinking: this way, we'd avoid the current repetition we have across different block repos.

Reply:
I like that idea. :-)

Reply:
> I'm wondering if it makes sense to move this function over to @wordpress/blocks... 🤔 this way, we'd avoid the current repetition we have across different block repos.

IIUC, if we need this to handle new categories on sites where Gutenberg has not been updated yet ([as it handles the other way round automagically](https://github.com/WordPress/gutenberg/pull/23695#issuecomment-653730155)), we can't actually move it to a Gutenberg package, because on those sites that utility won't be available. AFAIR, Jetpack imports use packages that are currently available with the installed version of the Gutenberg plugin, not those that are defined in `package.json` (i.e. `@worpdress/blocks`). Pinging @jeherve here for help 🙏 

If the above is true and we end up dropping the Gutenberg PR, how about we make it into a [Calypso package](https://github.com/Automattic/wp-calypso/tree/master/packages#packages) instead?

Reply:
> IIUC, if we need this to handle new categories on sites where Gutenberg has not been updated yet (as it handles the other way round automagically), we can't actually move it to a Gutenberg package, because on those sites that utility won't be available

But isn't the `@wordpress/data` a package that can be installed separately (even though it lives within the Gutenberg repo)? Right now, we need this function in the following packages:

- https://github.com/Automattic/wpcom-blocks/pull/178
- https://github.com/Automattic/jetpack/pull/16363
- https://github.com/Automattic/wp-calypso/pull/43670
- https://github.com/Automattic/block-experiments/pull/110

Wouldn't it just be a matter of publishing a new version of `@wordpress/blocks` and then updating it on these packages? 

I'm fine abandoning that PR and moving it to a Calypso package if we find it's the best way, by the way :)

Reply:
Ah, sorry I read too fast, you actually mention here:

> AFAIR, Jetpack imports use packages that are currently available with the installed version of the Gutenberg plugin, not those that are defined in package.json (i.e. @worpdress/blocks). Pinging @jeherve here for help 

Let's wait for @jeherve's insights about that, then :)

Thanks!

Reply:
Have we considered a linting rule that checks the category in warning about new categories as well? Something like checking the category of the `registerBlockType()` and warning if we know it's a category that's not going to work with older gutenbergs?